internal/osutil/mount_linux.go

```diff
@@ -30,3 +43,30 @@ func IsMounted(baseDir string) (bool, error) {
 	}
 	return false, nil
 }
+
+// Mount attaches a filesystem accessible via the device node specified by source
+// to the specified baseDir. If not existing, baseDir will be created before
+// mounting the filesystem.
+func Mount(source, baseDir, fstype string, readOnly bool) error {
+	if err := os.MkdirAll(baseDir, 0755); err != nil {
+		return fmt.Errorf("cannot create directory %q: %w", baseDir, err)
+	}
+
+	flags := uintptr(0)
+	if readOnly {
+		flags |= unix.MS_RDONLY
+	}
+	if err := syscallMount(source, baseDir, fstype, flags, ""); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Unmount removes the attachment of the topmost filesystem mounted on baseDir.
+func Unmount(baseDir string) error {
+	syscallSync()
```

Comment:
Nice point, thanks! I've briefly read about the `umount(8)` command handling sync automatically, but not about the `umount(2)` syscall doing so. Does the kernel really handle this? If so, you are completely right and we should remove this. If not, we'll also need to investigate whether a call to `mount(2)` with `MS_RDONLY` will sync previous changes to that same filesystem (and make sure for the cases we're not ro-remounting there's a way to lock the filesystem in order to prevent a race condition).